### ISSUE TITLE
find_sec_bugs.HARD_CODE_KEY-4 returns false positives

### ISSUE THREAD

**Issue by GiantCrocodile** [returntocorp/sgrep-rules#3568]
**Describe the bug**
The rule `find_sec_bugs.HARD_CODE_KEY-4` returns false-positives and seems quite useless in the way it's coded right now. See https://semgrep.dev/r?q=gitlab.find_sec_bugs.HARD_CODE_KEY-4.

**To Reproduce**
This legacy Java code is flagged in 3rd line:

```java
	public void validateCurrentPassword(FacesContext context, UIComponent component, Object value) {
		String confirmPassword = (String) value;
		if (!confirmPassword.equals(loggedInUser.getPassword())) {
			FacesMessage msg = new FacesMessage("Password does not match!");
			msg.setSeverity(FacesMessage.SEVERITY_ERROR);
			throw new ValidatorException(msg);
		}
	}
```

**Expected behavior**
No finding for this.

**Priority**
How important is this to you?
- [ ] P0: blocking me from making progress
- [ ] P1: this will block me in the near future
- [x] P2: annoying but not blocking me

**Screenshots**

**Additional context**
managed scan from Semgrep cloud

**Comment by LewisArdern** [comment]
@GiantCrocodile please raise an issue with https://gitlab.com/gitlab-org/security-products/sast-rules we do not manage gitlab rules

**Comment by GiantCrocodile** [comment]
I reported it here after my account got unstuck by GitLab team and after finding out how to post to their issue tracker as there is some weird workflow involved: https://gitlab.com/gitlab-org/gitlab/-/issues/566816
